bigtable/bigtable-proxy/src/main/java/com/google/cloud/bigtable/examples/proxy/core/CallLabels.java

```diff
@@ -155,4 +161,68 @@ private static Optional<String> extractAppProfileId(String[] encodedKvPairs) {
   private static String percentDecode(String s) {
     return URLDecoder.decode(s, StandardCharsets.UTF_8);
   }
+
+  @AutoValue
+  public abstract static class PrimingKey {
+    abstract Map<String, String> getMetadata();
+
+    abstract String getName();
+
+    abstract Optional<String> getAppProfileId();
+
+    public static Optional<PrimingKey> from(CallLabels labels) throws ParsingException {
+      Optional<String> resourceName = labels.getResourceName();
+      if (resourceName.isEmpty()) {
+        return Optional.empty();
+      }
+      String[] resourceNameParts = resourceName.get().split("/", 5);
+      if (resourceNameParts.length < 4
+          || !resourceNameParts[0].equals("projects")
+          || !resourceNameParts[2].equals("instances")) {
+        return Optional.empty();
+      }
+      String instanceName =
+          "projects/" + resourceNameParts[1] + "/instances/" + resourceNameParts[3];
+      StringBuilder reqParams =
+          new StringBuilder()
+              .append("name=")
+              .append(URLEncoder.encode(instanceName, StandardCharsets.UTF_8));
+
+      Optional<String> appProfileId = labels.getAppProfileId();
+      appProfileId.ifPresent(val -> reqParams.append("app_profile_id=").append(val));
```

Comment:
should this be "&app_profile_id="?

Reply:
yep, I fixed it in a later PR, but backported it now to make this PR complete